src/middlewared/middlewared/plugins/idmap_/idmap_sss.py

```diff
@@ -0,0 +1,223 @@
+
+        `getidbysid`
+        {'S-1-5-21-3696504179-2855309571-923743039-1020': {'id': 565200020, 'type': 1}}
+
+        Sample of what we return:
+        {
+            'id_type': 'GROUP',
+            'id': 565200020,
+            'name': 'smbuser',
+            'sid': 'S-1-5-21-3696504179-2855309571-923743039-1020'
+        }
+        """
+        if not (sid_entry := sssclient.getsidbygroupname(groupname)):
+            return None
+
+        sid = sid_entry[groupname]['sid']
+        id_type = sid_entry[groupname]['type']
+
+        if not (id_entry := sssclient.getidbysid(sid)):
+            return None
+
+        return {
+            'id_type': IDType(id_type).name,
+            'id': id_entry[sid]['id'],
+            'name': groupname,
+            'sid': sid
+        }
+
+    def _gid_to_entry(self, gid):
+        """ convert gid to idmap entry dict -- see above _groupname_to_entry()"""
+        if not (sid_entry := sssclient.getsidbygid(gid)):
+            return None
+
+        sid = sid_entry[gid]['sid']
+        id_type = sid_entry[gid]['type']
+
+        if not (name_entry := sssclient.getnamebysid(sid)):
+            return None
+
+        return {
+            'id_type': IDType(id_type).name,
+            'id': gid,
+            'name': name_entry[sid]['name'],
+            'sid': sid
+        }
+
+    def _uid_to_entry(self, uid):
+        """ convert gid to idmap entry dict -- see above _username_to_entry()"""
+        if not (sid_entry := sssclient.getsidbyuid(uid)):
+            return None
+
+        sid = sid_entry[uid]['sid']
+        id_type = sid_entry[uid]['type']
+
+        if not (name_entry := sssclient.getnamebysid(sid)):
+            return None
+
+        return {
+            'id_type': IDType(id_type).name,
+            'id': uid,
+            'name': name_entry[sid]['name'],
+            'sid': sid
+        }
+
+    def _sid_to_entry(self, sid):
+        """ convert sid to idmap entry dict -- see above _username_to_entry()"""
+        if not (id_entry := sssclient.getidbysid(sid)):
+            return None
+
+        if not (name_entry := sssclient.getnamebysid(sid)):
+            return None
+
+        return {
+            'id_type': IDType(id_entry[sid]['type']).name,
+            'id': id_entry[id_entry[sid]]['id'],
+            'name': name_entry[sid]['name'],
+            'sid': sid
+        }
+
+    def sids_to_idmap_entries(self, sidlist):
+        """
+        Bulk conversion of list of sids to idmap entries
+
+        sample output:
+        {
+          "mapped": {
+            "S-1-5-21-3696504179-2855309571-923743039-1020": {
+              "id_type": "USER",
+              "id": 565200020,
+              "name": "smbuser",
+              "sid": "S-1-5-21-3696504179-2855309571-923743039-1020"
+            }
+          }
+          "unmapped": {
+            "S-1-5-21-3696504179-2855309571-923743039-1022": "S-1-5-21-3696504179-2855309571-923743039-1020"
```

Comment:
Presumably the key and value should be the same value.  (Currently the last digit differs).